lib/elixir/src/elixir.erl

```diff
@@ -61,6 +61,7 @@ start(_Type, _Args) ->
     {docs, true},
     {ignore_already_consolidated, false},
     {ignore_module_conflict, false},
+    {on_undefined_variable, raise},
```

Comment:
Perhaps it would be nicer to default it to `warn` globally and set it to `raise` in a smaller scope? While this is great for **my** code, I don't think compiling dependencies with this option gives significant benefit. Instead, it potentially causes issues by not compiling correctly-functioning dependencies.

Reply:
> it potentially breaks the compilation of correctly functioning dependencies.

It will only break dependencies that have been emitting warnings for 5+ years (almost 6). The only reason we are making this a flag is because we want to provide an option for those dependencies to work, but raising is the default we expect. :)